filters/NormalFilter.cpp

```diff
@@ -58,33 +58,41 @@ std::string NormalFilter::getName() const
     return s_info.name;
 }
 
-
 void NormalFilter::addArgs(ProgramArgs& args)
 {
     args.add("knn", "k-Nearest Neighbors", m_knn, 8);
+    m_vxArg = &args.add("vx", "Viewpoint X coordinate", m_vx);
+    m_vyArg = &args.add("vy", "Viewpoint Y coordinate", m_vy);
+    m_vzArg = &args.add("vz", "Viewpoint Z coordinate", m_vz);
 }
 
-
 void NormalFilter::addDimensions(PointLayoutPtr layout)
 {
     using namespace Dimension;
 
-    layout->registerDims({Id::NormalX, Id::NormalY, Id::NormalZ,
-        Id::Curvature});
+    layout->registerDims(
+        {Id::NormalX, Id::NormalY, Id::NormalZ, Id::Curvature});
 }
 
-
 void NormalFilter::doFilter(PointView& view)
 {
     m_knn = 8;
     filter(view);
 }
 
-
 void NormalFilter::filter(PointView& view)
 {
     KD3Index& kdi = view.build3dIndex();
 
+    BOX3D bounds;
```

Comment:
We are discussing alternatives in #1638, so this will all be refactored in a bit.